### PR TITLE
Update PyTorch example to illustrate calling .backward() on a non-scalar

### PR DESCRIPTION
**Pull request by NishantTharani**
I thought it would be useful to illustrate how calling .backward() on a non-scalar is also possible, as long as you pass in an appropriate `gradient` argument, which is simply all ones in our case.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1090/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1090/

**Comment by NishantTharani** [comment]
Hi, this is literally my first attempt at contributing to an open source project ever. I saw a test failed but I'm having difficulty seeing how the test output relates to my contribution. Could anyone help assess my contribution please?

File: chapter_preliminaries/autograd.md [hunk 1]
```diff
@@ -211,9 +211,14 @@ x.grad  # Equals to y = sum(x * x)
...
 #@tab pytorch
+# Invoking `backward` on a non-scalar requires passing in a `gradient` argument
+# which specifies the gradient of the differentiated function w.r.t `self`.
+# In our case, we simple want to sum the partial derivatives, so passing
+# in a gradient of ones is appropriate.
 x.grad.zero_()
 y = x * x
-y.sum().backward()  # `backward` only supports for scalars
+# y.sum().backward() equivalent to the below
+y.backward(torch.ones(len(x)))
```
Comment (AnirudhDagar) [inline chapter_preliminaries/autograd.md:221]:
```suggestion
# y.backward(torch.ones(len(x))) equivalent to the below
y.sum().backward()
```

File: chapter_preliminaries/autograd.md [hunk 2]
```diff
@@ -211,9 +211,14 @@ x.grad  # Equals to y = sum(x * x)
 
 ```{.python .input}
 #@tab pytorch
+# Invoking `backward` on a non-scalar requires passing in a `gradient` argument
+# which specifies the gradient of the differentiated function w.r.t `self`.
+# In our case, we simple want to sum the partial derivatives, so passing
```
Comment (AnirudhDagar) [inline chapter_preliminaries/autograd.md:216]:
```suggestion
# In our case, we simply want to sum the partial derivatives, so passing
```

File: chapter_preliminaries/autograd.md [hunk 3]
```diff
@@ -211,9 +211,14 @@ x.grad  # Equals to y = sum(x * x)
...
+# in a gradient of ones is appropriate.
```
Comment (AnirudhDagar) [inline chapter_preliminaries/autograd.md:217]: No period at the end of comments.
```suggestion
# in a gradient of ones is appropriate
```

**Comment by NishantTharani** [comment]
Thanks @AnirudhDagar !

**Comment by mli** [comment]
Job d2l-en/PR-1090/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1090/

**Comment by mli** [comment]
Job d2l-en/PR-1090/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1090/
